### PR TITLE
v4 upgrade handler

### PR DESCRIPTION
**Pull request by EricBolten**
Initial use of the upgrade module to move us to v4. Adds the cork and cellarfees module, and will migrate the gravity module from v1 to v2.

Opening this up as a draft PR as the gravity dependency doesn't have a released version yet.

### CODE REVIEW

**Comment by zmanian** [comment]
I think we need something in the upgrade handler to delete the store for Allocation module...

**Comment by EricBolten** [comment]
> I think we need something in the upgrade handler to delete the store for Allocation module...

I believe this bit will cause the allocation store to be deleted:

https://github.com/PeggyJV/sommelier/blob/31a622ad016dd209574a678a2355e2d6dbdf7d67/app/app.go#L730-L737

File: app/upgrades/v4/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+## Summary of changes
+
+* Switch to use of the upgrade module
+* Add the cork module for receiving arbitrary logic calls
+* Add the cellarfees module to provide a module account for cellar fees to be bridged to
+* Support sending to specified module accounts over the bridge
+* Community spend governance proposal for sending funds over the bridge to an Ethereum address
+* Fix a bug affecting the capitalization of ERC20 addresses in denominations
+* Fix a bug incorrectly setting the timeouts of ContractCallTxs
```
Comment (mvid) [inline app/upgrades/v4/README.md:13]: Do we document the removal of allocation? Or does it not matter because it was never used?

**Review by mvid — APPROVED** [review]

File: app/upgrades/v4/README.md (hunk 1 above)
Comment (EricBolten) [reply app/upgrades/v4/README.md:13]: Good note, I'll add it to the README.

**Comment by EricBolten** [comment]
Integration tests pass locally with an orchestrator image built from gravity-bridge main, the CI test is pulled the latest tag though which won't be updated until there's a gravity release.

**Review by zmanian — APPROVED** [review]

**Pull request merged**